### PR TITLE
DEV: Various A11Y improvements for the new user menu

### PR DESCRIPTION
**Pull request by OsamaSayegh**
This PR includes various accessibility improvements for the new user menu:

* Add `title` attributes to the user menu tabs
* Properly label lists (by adding `aria-labelledby` to `<ul>` elements) for screen readers
* Change the user menu structure so that the tabs come before the content panel in the DOM, but use CSS to reverse them visually.
  Normally, changing the order of elements via CSS is bad for accessibility, but I believe this is one of the rare scenarios where it [makes sense](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Ordering_Flex_Items#use_cases_for_order). Prior to this change, if you want to reach the first notification item after you select a tab using the keyboard, you have to hit <kbd>ctrl</kbd>+<kbd>tab</kbd> because the notifications list is before the tabs list. However, with this change, <kbd>tab</kbd> will move you to the first item in the list after you select a tab using your keyboard.
* Aria-hide the unread notifications badge/count on the tabs because the `title` attribute on the tab indicates the unread count.
* Add some tests.

### CODE REVIEW

File: app/assets/javascripts/discourse/tests/test-boot-ember-cli.js [hunk 1]
```diff
@@ -48,7 +48,8 @@ document.addEventListener("discourse-booted", () => {
   if (QUnit.config.seed === undefined) {
     // If we're running in browser, default to random order. Otherwise, let Ember Exam
     // handle randomization.
-    QUnit.config.seed = Math.random().toString(36).slice(2);
+    // QUnit.config.seed = Math.random().toString(36).slice(2);
+    QUnit.config.seed = "9atcpy4t344";
```
Comment (pmusaraj) [inline app/assets/javascripts/discourse/tests/test-boot-ember-cli.js:52]: Did you mean to commit this?

File: config/locales/client.en.yml [hunk 2]
```diff
@@ -2567,10 +2567,45 @@ en:
...
-      sr_menu_tabs: "Menu tabs"
+      sr_menu_tabs: "User menu tabs"
       view_all_notifications: "view all notifications"
       view_all_bookmarks: "view all bookmarks"
       view_all_messages: "view all personal messages"
+      tabs:
+        all_notifications: "All notifications"
+        replies: "Replies"
+        replies_with_unread:
+          one: "Replies - %{count} unread"
```
Comment (pmusaraj) [inline config/locales/client.en.yml:2578]: Would it make sense to use `%{count} unread reply` here (and in similar cases below)?

**Review by pmusaraj — COMMENTED** [review]
Looks good, just two minor comments.

File: config/locales/client.en.yml (hunk 2 above)
Comment (OsamaSayegh) [reply config/locales/client.en.yml]: sure, let's try that!

File: config/locales/client.en.yml [hunk 3]
```diff
@@ -2567,10 +2567,45 @@ en:
...
+          one: "Personal messages - %{count} unread message"
+          other: "Personal messages - %{count} unread messages"
+        bookmarks: "Bookmarks"
+        bookmarks_with_unread:
+          one: "Bookmarks - %{count} unread bookmark"
+          other: "Bookmarks - %{count} unread bookmarks"
+        review_queue: "Review queue"
+        review_queue_with_unread:
+          one: "Review queue - %{count} needs review"
+          other: "Review queue - %{count} need review"
```
Comment (pmusaraj) [inline config/locales/client.en.yml:2603]:
```suggestion
          one: "Review queue - %{count} item needs review"
          other: "Review queue - %{count} items need review"
```

**Review by pmusaraj — APPROVED** [review]

**Pull request merged**